manager/client/main.go

```diff
@@ -0,0 +1,103 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	rpc_license "github.com/enfabrica/enkit/manager/rpc"
+	grpcCodes "google.golang.org/grpc/codes"
+	grpcStatus "google.golang.org/grpc/status"
+	"google.golang.org/grpc"
+	"io"
+	"log"
+	"os"
+	"os/exec"
+	"os/user"
+	"strings"
+	"time"
+)
+
+func run(timeout time.Duration, cmd string, args ...string) {
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+    defer cancel()
+    job := exec.CommandContext(ctx, cmd, args...)
+    err := job.Run()
+    if ctx.Err() == context.DeadlineExceeded {
+		log.Fatalf("Job failed to complete after %d seconds: %s %s \n", timeout, cmd, strings.Join(args, " "))
```

Comment:
genuinely asking: will the log.Fatal actually kill the runaway child process? eg, I would not expect the **go** process exiting to kill its children. I would have expected a `job.Process.Kill()` to be necessary to ensure it is actually killed before the log.Fatal.

Reply:
`ctx.Done()` is called internally after the timeout error
https://stackoverflow.com/questions/61001275/is-it-safe-to-leave-out-ctx-done-when-using-the-go-context-package